### PR TITLE
Upgrade dependencies to address cryptography security vulnerability

### PR DESCRIPTION
**Pull request by 97nitt**
Get `cryptography` upgraded to address security vulnerability.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -164,7 +167,7 @@ msal-extensions==1.1.0
     # via azure-identity
 mypy-extensions==1.0.0
     # via typing-inspect
-numpy==1.26.2
+numpy==1.24.4
```
Comment (walles2) [inline requirements.txt:170]: We're ok with downgrading Numpy and Pandas here?

Comment (97nitt) [reply requirements.txt:170]: This is what the dependency resolver came up with 🤷 

I was a little concerned, more with the possibility of introducing a new vulnerability, but `docker scout` scans of the agent images (ran locally) didn't come up with anything new.

Comment (97nitt) [reply requirements.txt:170]: @walles2 @mrostan any concerns here with the downgraded `numpy` and `pandas`?

Comment (mrostan) [reply requirements.txt:170]: I think it's fine, data-collector in master is using:
- numpy==1.23.4
- pandas==1.3.5
so, even when we're downgrading we're still using versions after those being used by the dc

Comment (mrostan) [reply requirements.txt:170]: sorry, assumed it was already approved

**Review by mrostan — APPROVED** [review]

**Pull request merged**